packages/webamp-modern-2/src/skin/makiClasses/Movable.ts

```diff
@@ -0,0 +1,203 @@
+    layout.setResizing(
+      this._div.style.getPropertyValue("cursor"),
+      CURSOR,
+      CURSOR
+    );
+    const startX = downEvent.clientX;
+    const startY = downEvent.clientY;
+
+    const handleMove = (moveEvent: MouseEvent) => {
+      const newMouseX = moveEvent.clientX;
+      const newMouseY = moveEvent.clientY;
+      const deltaY = newMouseY - startY;
+      const deltaX = newMouseX - startX;
+      layout.setResizing("move", deltaX, deltaY);
+    };
+
+    const handleMouseUp = (upEvent: MouseEvent) => {
+      upEvent.stopPropagation();
+      if (upEvent.button != 0) return; // only care LeftButton
+      document.removeEventListener("mousemove", handleMove);
+      document.removeEventListener("mouseup", handleMouseUp);
+      const newMouseX = upEvent.clientX;
+      const newMouseY = upEvent.clientY;
+      const deltaY = newMouseY - startY;
+      const deltaX = newMouseX - startX;
+      layout.setResizing("final", deltaX, deltaY);
+    };
+    document.addEventListener("mousemove", handleMove);
+    document.addEventListener("mouseup", handleMouseUp);
+  }
+
+  _registerMovingEvents() {
+    if (this._movingEventsRegisterd) {
+      return;
+    }
+    this._movingEventsRegisterd = true;
+    this._div.addEventListener("mousedown", this._handleMoving.bind(this));
+  }
+
+  _unregisterMovingEvents() {
+    if (this._movingEventsRegisterd) {
+      this._div.removeEventListener("mousedown", this._handleMoving);
+      this._movingEventsRegisterd = false;
+    }
+  }
+
+  _handleMoving(downEvent: MouseEvent) {
+    downEvent.stopPropagation();
+    if (downEvent.button != 0) return; // only care LeftButton
+    const layout = this.getparentlayout() as Layout;
+    layout.setMoving("start", 0, 0);
+    const startX = downEvent.clientX;
+    const startY = downEvent.clientY;
+
+    const handleMove = (moveEvent: MouseEvent) => {
+      const newMouseX = moveEvent.clientX;
+      const newMouseY = moveEvent.clientY;
+      const deltaY = newMouseY - startY;
+      const deltaX = newMouseX - startX;
+      layout.setMoving("move", deltaX, deltaY);
+    };
+
+    const handleMouseUp = (upEvent: MouseEvent) => {
+      if (upEvent.button != 0) return; // only care LeftButton
+      upEvent.stopPropagation();
+      document.removeEventListener("mousemove", handleMove);
+      document.removeEventListener("mouseup", handleMouseUp);
+      const newMouseX = upEvent.clientX;
+      const newMouseY = upEvent.clientY;
+      const deltaY = newMouseY - startY;
+      const deltaX = newMouseX - startX;
+      layout.setMoving("final", deltaX, deltaY);
+    };
+    document.addEventListener("mousemove", handleMove);
+    document.addEventListener("mouseup", handleMouseUp);
+  }
+
+  draw() {
+    super.draw();
+    //   this._div.style.pointerEvents = this._movable || this._resizable? "auto": "none";
+    if (this._movable || this._resizable) {
+      // this._div.style.removeProperty('pointer-events');
```

Comment:
Do we need to remove `--pointer-events-by` here?

Reply:
no. can not be.
Because the pointerEvent (CSS property) is inherance, 
hence it should be explicited, to prevent of in case the element's parent has `pointer-event: none;`.